packages/github/src/github.ts

```diff
@@ -17,11 +23,90 @@ export const context = new Context.Context()
 export class GitHub extends Octokit {
   graphql: GraphQL
 
-  constructor(token: string, opts: Omit<Octokit.Options, 'auth'> = {}) {
-    super({...opts, auth: `token ${token}`})
+  /* eslint-disable no-dupe-class-members */
+  // Disable no-dupe-class-members due to false positive for method overload
+  // https://github.com/typescript-eslint/typescript-eslint/issues/291
 
-    this.graphql = graphql.defaults({
-      headers: {authorization: `token ${token}`}
-    })
+  constructor(token: string, opts?: Octokit.Options)
```

Comment:
I made a change to allow users to specify their own auth settings (in opts) instead of providing a token.

[Customers who are using other authentication methods](https://github.com/actions/toolkit/issues/106) should still be able to use this class for the proxy benefits.

However I'm not sure what to do yet about graphql. Since the opts parameter is specifically `Octokit.options`.

I need to look closer tomorrow and see if setting graphql `defaults.headers.authorization` equal to octokit `opts.auth` will work.